From 135afc6ed141416dc71740c60a07937af2b3e9fb Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinh.nguyen@intel.com>
Date: Thu, 26 Aug 2021 09:36:03 -0500
Subject: [PATCH 38/42] clk: socfpga: agilex: fix up s2f_user0_clk
 representation

commit b195cf0aa9337be383125e30824193698cb59e3e from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

[upstream commit f817c132db679d492d96c60993fa2f2c67ab18d0]

Correct the s2f_user0_mux clock representation.

Fixes: 80c6b7a0894f ("clk: socfpga: agilex: add clock driver for the Agilex platform")
Cc: stable@vger.kernel.org
Signed-off-by: Kris Chaplin <kris.chaplin@intel.com>
Signed-off-by: Dinh Nguyen <dinguyen@kernel.org>
Link: https://lore.kernel.org/r/20210713144621.605140-2-dinguyen@kernel.org
Signed-off-by: Stephen Boyd <sboyd@kernel.org>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/clk/socfpga/clk-agilex.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/socfpga/clk-agilex.c b/drivers/clk/socfpga/clk-agilex.c
index ab86e98fc628..571fa7b3eaf2 100644
--- a/drivers/clk/socfpga/clk-agilex.c
+++ b/drivers/clk/socfpga/clk-agilex.c
@@ -58,6 +58,7 @@ static const char * const emac_mux[] = {"emaca_free_clk", "emacb_free_clk",
 					"boot_clk"};
 static const char * const noc_mux[] = {"noc_free_clk", "boot_clk"};
 static const char * const sdmmc_mux[] = {"sdmmc_free_clk", "boot_clk"};
+static const char * const s2f_user0_mux[] = {"s2f_user0_free_clk", "boot_clk"};
 static const char * const s2f_user1_mux[] = {"s2f_user1_free_clk", "boot_clk"};
 static const char * const psi_mux[] = {"psi_ref_free_clk", "boot_clk"};
 static const char * const gpio_db_mux[] = {"gpio_db_free_clk", "boot_clk"};
@@ -159,6 +160,8 @@ static const struct stratix10_gate_clock agilex_gate_clks[] = {
 	  4, 0x98, 0, 16, 0x88, 3, 0},
 	{ AGILEX_SDMMC_CLK, "sdmmc_clk", NULL, sdmmc_mux, ARRAY_SIZE(sdmmc_mux), 0, 0x7C,
 	  5, 0, 0, 0, 0x88, 4, 4},
+	{ AGILEX_S2F_USER0_CLK, "s2f_user1_clk", NULL, s2f_user0_mux, ARRAY_SIZE(s2f_user0_mux), 0, 0x24,
+	  6, 0, 0, 0, 0x30, 2, 0},
 	{ AGILEX_S2F_USER1_CLK, "s2f_user1_clk", NULL, s2f_user1_mux, ARRAY_SIZE(s2f_user1_mux), 0, 0x7C,
 	  6, 0, 0, 0, 0x88, 5, 0},
 	{ AGILEX_PSI_REF_CLK, "psi_ref_clk", NULL, psi_mux, ARRAY_SIZE(psi_mux), 0, 0x7C,
-- 
2.31.1

